### PR TITLE
Save results in csv

### LINKED ISSUE #14: Save output data in a csv file
Also create a folder structure for results

### LINKED ISSUE #19: Save individual runs info
For every transpile run, also save:
- seed
- circuit depth
- gate count

Another option is to have a separate detailed output file, in addition to the main ave/stdev results submitted to metriq.

### PR DESCRIPTION
**Pull request by karlaspuldaro**
Closes #14 
Closes #19 

This PR:
-  Stores experiment run detailed results in csv files, filenames organized by qasm, architecture and qiskit version
- Uses pandas for data manipulation
- Data Columns are: **Qasm file|Method|Date|Opt level|Platform|Seed|Circuit depth|Gate count**
- `Date` is qiskit-terra's version release date used in the experiment
- Mean values are rounded to int, standard deviation values are rounded to up to 3 decimals
- Separates data collection from the data processing, the final result is printed to screen
- Installs [metriq-client](https://github.com/unitaryfund/metriq-client) package and dependencies

Next (**Coming on separate PR**): #9 - write a post processing script with data actually submitted to the metriq API

CSV output examples: 
ex1_226-qiskit0.24.1-rigetti_16q_aspen.csv
image
ex1_226-qiskit0.24.1-ibm_rochester.csv
image

Post processing results output:
image

### CODE REVIEW

**Comment by karlaspuldaro** [comment]
Historical submission - https://metriq.info/Submission/595 
(Needs editing)

**Comment by 1ucian0** [comment]
I think the csv keeps redundant information. The goal of the file is to split the data collection from the data processing.

Each "experiment" (a compilation, in this case) is a single csv entry per run with, say, the following columns:
 * qasm file
 * qiskit version
 * platform
 * seed
 * optimization level
 * circuit depth
 * gate count
 
once all the experiments run, it is possible to use this table to build the results (the data processing stage) to submit, with the average and standard dev.

**Comment by karlaspuldaro** [comment]
Thank you @1ucian0 !
I addressed your suggestions and updated the PR description and examples accordingly. 
This is ready for another review.

File: tox.ini [hunk 1]
```diff
@@ -12,5 +12,10 @@ deps =
     requests
     pyzx
     pandas
-commands = 
+    tea-client==0.0.7
+    tea-console==0.0.5
+    typer==0.3.2
+commands =
+    pip install --upgrade https://github.com/unitaryfund/metriq-client/tarball/master
```
Comment (karlaspuldaro) [inline tox.ini:19]: Adding metriq-client dependencies and installing package from github
@WrathfulSpatula couldn't find a pypi module distribution for it, so installed the package in our env this way, and was able to create and run a`client.hello()` successfully :)
Pls let us know if there is a better installation method

Comment (WrathfulSpatula) [reply tox.ini:19]: @karlaspuldaro The client needs some "TLC" before a PyPi release, but we're working on it! This looks like the right idea, for the current state of development. Glad you got it working!

**Pull request merged**